api/Validators/SupplierValidator.cs

```diff
@@ -0,0 +1,17 @@
+using FluentValidation;
+
+public class SupplierValidator : AbstractValidator<Supplier>
+{
+    public SupplierValidator()
+    {
+        RuleFor(Supplier => Supplier.Name)
+            .NotNull().WithMessage("name is required.")
+            .NotEmpty().WithMessage("name cannot be empty.");
+        RuleFor(Supplier => Supplier.Code)
+            .NotNull().WithMessage("code is required.")
+            .NotEmpty().WithMessage("code cannot be empty.");
+        RuleFor(Supplier => Supplier.Reference)
+            .NotNull().WithMessage("reference is required.")
+            .NotEmpty().WithMessage("reference cannot be empty.");
+    }
+}
```

Comment:
Ik mis hier de validation van address id en contact id als ze uberhaupt bestaan in de DB